### PR TITLE
stub cellVoice

### PR DESCRIPTION
**Pull request by Megamouse**
Added error_code and arguments and some reversed error checks to cellVoice.

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellVoice.h [hunk 1]
```diff
@@ -60,32 +45,139 @@ enum
...
+		struct
+		{
+			u32 bufSize;
+			CellVoicePCMFormat format;
+		} pcmaudio;
+		struct
+		{
+			u32 playerId;
+		} device;
+	};
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellVoice.h]: add be_t<> to bitrate, format and playerId.

File: rpcs3/Emu/Cell/Modules/cellVoice.h [hunk 2]
```diff
@@ -1,12 +1,14 @@
-#pragma once
-
+﻿#pragma once
 
+using sys_event_queue_t = u32;
+using sys_memory_container_t = u32;
+using sys_ipc_key_t = u64;
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellVoice.h]: We don't really use those types even in code which it was originated from.

File: rpcs3/Emu/Cell/Modules/cellVoice.h [hunk 3]
```diff
@@ -60,32 +45,139 @@ enum
...
+	CellVoiceEventType eventMask; // TODO: correct type ?
+	CellVoiceVersionCheck version; // TODO: correct type ?
+	be_t<s32> appType;
+	u8 reserved[32 - sizeof(s32) * 3];
+};
+
+struct CellVoicePCMFormat
+{
+	be_t<u8> numChannels;
+	be_t<u8> sampleAlignment;
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellVoice.h]: u8 doesn't need be_t<>, BE stands for Big-Endian byte ordering and the size of u8 is exactly one byte, so ordering is irrelevant here.

File: rpcs3/Emu/Cell/Modules/cellVoice.h [hunk 4]
```diff
@@ -60,32 +45,139 @@ enum
...
+	CellVoicePcmDataType dataType; // TODO: correct type ?
+	CellVoiceSamplingRate sampleRate; // TODO: correct type ?
+};
+
+struct CellVoicePortParam // aligned(64)
+{
+	CellVoicePortType portType; // TODO: correct type ?
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellVoice.h]: Add be_t<> for all enums.

File: rpcs3/Emu/Cell/Modules/cellVoice.h [hunk 5]
```diff
@@ -60,32 +45,139 @@ enum
...
+		struct
+		{
+			u32 playerId;
+		} device;
+	};
+};
+
+struct CellVoiceStartParam // aligned(32)
+{
+	sys_memory_container_t container; // TODO: correct type ?
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellVoice.h]: This is basically u32, also needs be_t<>.

**Comment by zeus128** [comment]
God of War: Ascension™ "Prison of the Damned" Demo [NPEA90123]

U {PPU[0x100000f] Thread (voice thread) [0x009c5500]} cellVoice TODO: cellVoiceGetPortInfo(portId=0, pInfo=*0xd0063e00

File: rpcs3/Emu/Cell/Modules/cellVoice.h (hunk 1 above)
Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellVoice.h]: bufSize not?

Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellVoice.h]: yes I meant bufSize, not format

File: rpcs3/Emu/Cell/Modules/cellVoice.cpp [hunk 6]
```diff
@@ -1,220 +1,424 @@
...
+		return CELL_VOICE_ERROR_LIBVOICE_NOT_INIT;
+
 	return CELL_OK;
 }
 
-s32 cellVoiceRemoveNotifyEventQueue()
+error_code cellVoiceRemoveNotifyEventQueue(u64 key)
 {
-	UNIMPLEMENTED_FUNC(cellVoice);
+	cellVoice.todo("cellVoiceRemoveNotifyEventQueue(key=%d)", key);
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellVoice.cpp]:
```suggestion
	cellVoice.todo("cellVoiceRemoveNotifyEventQueue(key=0x%llx)", key);
```

File: rpcs3/Emu/Cell/Modules/cellVoice.cpp [hunk 7]
```diff
@@ -1,220 +1,424 @@
...
+		return CELL_VOICE_ERROR_LIBVOICE_NOT_INIT;
+
 	return CELL_OK;
 }
 
-s32 cellVoiceSetNotifyEventQueue()
+error_code cellVoiceSetNotifyEventQueue(u64 key, u64 source)
 {
-	UNIMPLEMENTED_FUNC(cellVoice);
+	cellVoice.todo("cellVoiceSetNotifyEventQueue(key=%d, source=%d)", key, source);
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellVoice.cpp]:
```suggestion
	cellVoice.todo("cellVoiceSetNotifyEventQueue(key=0x%llx, source=%d)", key, source);
```

Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellVoice.cpp]: source is also u64, didnt notice.

**Pull request merged**